src/controllers/middlewares/addPlan.js

```
@@ -0,0 +1,37 @@
+const { sign } = require('jsonwebtoken');
+const insertPlan = require('../../database/queries/insertPlan');
+const insertCategoryPlan = require('../../database/queries/insertCategoryPlan');
+
+exports.addPlan = (req, res, next) => {
+  const { income, startDate, endDate } = req.session;
+  const { id } = req.token;
+  insertPlan(id, income, startDate, endDate)
+    .then((plans) => {
+      if (plans.rowCount === 1) {
+        req.token.planId = plans.rows[0].id;
+        next();
+      }
```

@IsraaSulaiman You could return an error in an `else` for this if-statement, just to cover all scenarios.